scio-parquet/src/main/scala/com/spotify/scio/parquet/types/ParquetTypeIO.scala

```diff
@@ -69,6 +65,11 @@ final case class ParquetTypeIO[T: ClassTag: Coder: ParquetType](path: String) ex
       .withKeyTranslation(new SimpleFunction[Void, JBoolean]() {
         override def apply(input: Void): JBoolean = true
       })
+      .withValueTranslation(new SimpleFunction[T, T]() {
+        override def apply(input: T): T = input
+
+        override def getInputTypeDescriptor: TypeDescriptor[T] = TypeDescriptor.of(cls)
+      }, CoderMaterializer.beam(sc, coder))
```

Comment:
you should be able to just write `CoderMaterializer.beam(sc, Coder[T])` without the implicit summons on line 49, since ParquetTypeIO has context bound `T: Coder`

otherwise, this PR makes sense, thanks for surfacing this! I just want to look into HadoopFormatIO + do some manual verification of this before merging.

Reply:
+1, the same approach is actually used in the [ParquetAvroIO](https://github.com/spotify/scio/blob/73d731d99bef38d1eefcfd5c89e3a44648550ec4/scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/ParquetAvroIO.scala#L226) so it makes a lot of sense to do the same. Would be interesting to see some read benchmarks w/ and w/o the fix!

Reply:
Executed a sample job w/ and w/o coder override and observed difference in the total vCPU hrs:

* 8.764 vCPU hr - w/o coder override
* 3.747 vCPU hr - w/ coder fix

which is quite significant - about 2.3x times better with the fix